### DISCUSSION TITLE
HTML + Xpath (webscrapping) with JavaScript

### DISCUSSION THREAD

**Discussion by taccle** [FreshRSS/FreshRSS#6891] [Q&A] [answered]
Hello,

I'm trying to add a new feed from https://jobs.volvogroup.com/?locale=fr_FR/ with Xpath (Web scraping) with `//div[@id="job--table"]` for "finding new items" and `descendant::span[@data-id="title"]` for item title.
I'm getting the following error "Blast! This feed has encountered a problem. Please verify that it is always reachable then update it." In the logs, I can only retrieve "HTML+XPath Web scraping failed for [https://jobs.volvogroup.com/?locale=fr_FR/]"

I initially thought it was because FreshRSS couldn't access the website. But when I run curl https://jobs.volvogroup.com/?locale=fr_FR/, I properly get the page source and I see that the job offers are not present. I suppose this is because the Xpath I'm using only appears with JavaScript as discussed in this thread https://github.com/FreshRSS/FreshRSS/discussions/6876. 

I didn't find any discussion on how I could scrap such website. Do you have any hints on this?

Thanks in advance for your help.

**Comment by Alkarex** [comment, marked as answer]
Hello,
In your case, the actual relevant content is provided by `https://jobs.volvogroup.com/feed/361555` (easiest to see in your Web browser network tools), which you should be able to read into FreshRSS using the `XML+XPath` mode.

**Reply by taccle** [reply]
Arf, it wasn't so complicated in the end. Thank you for your support and for your work on FreshRss!

**Comment by taccle** [comment]
Hello,

I'm coming back on this discussion because I'm trying to scrap many job offers websites and I'm facing again the same issue as before. And in this case I cannot find neither the job offers, neither a "feed" link in the page source.
Is there any list of tips to find a feed in a webpage? What I do up to now is to search:
- for a RSS logo
- for "feed" / "rss" in the page source
- for the content I need in the page source (either as html or json)


Is it possible (maybe not directly with FreshRSS) to get the page after JavaScript transformation? Is there any ressources or discussions explaining how to do it? If not, would a freshrss extension such as [freshrss-flaresolverr-extension](https://github.com/ravenscroftj/freshrss-flaresolverr-extension) would somehow do the trick (for example with an implementation of Splash instead of flaresolverr)?
I leave here the websites I'm unable to scrap (but in the end, I'm more interested for answers of my questions above than to actually scrap those websites):
- https://careers.se.com/global/jobs 
- https://jobs.continental.com/en/#/ (actually they do have rss but not for job offers)


Thanks in advance for your help.

**Reply by Frenzie** [reply]
> neither a "feed" link in the page source.

If one's there FreshRSS will usually find it if you simply give the URL btw.

But in this case the data is loaded from `https://careers.se.com/api/jobs?page=1&sortBy=relevance&descending=false&internal=false&deviceId=undefined&domain=se.jibeapply.com`

You can find out by going to the sources tab (Chrome) or network tab (Firefox) in the inspector, press Ctrl+Shift+F and search for something like `country sales`.

> Is it possible (maybe not directly with FreshRSS) to get the page after JavaScript transformation? Is there any ressources or discussions explaining how to do it?

Possible sure, with Selenium for example. See something like https://pythonbasics.org/selenium-get-html/ but with something like `html = driver.execute_script("return document.documentElement.outerHTML")` instead. But I doubt it's worth the trouble compared to visiting the site once a week.

> If not, would a freshrss extension such as [freshrss-flaresolverr-extension](https://github.com/ravenscroftj/freshrss-flaresolverr-extension) would somehow do the trick (for example with an implementation of Splash instead of flaresolverr)?

Oh yeah, that does do something conceptually very similar to what I just said, just significantly more complicated in order to deal with Cloudflare.

> https://jobs.continental.com/en/#/ (actually they do have rss but not for job offers)

I'll leave finding it as an exercise. You can find it swiftly using the steps I described above: go to the sources/network tab, press Ctrl+Shift+F, type something like `control business`.

**Reply by taccle** [reply]
Thanks a lot, this technique with the network tab seems to be pretty effective! I think that'll be more than enough for my use case.
